src/support/err.c

```diff
@@ -504,12 +502,11 @@ __wt_illegal_value(WT_SESSION_IMPL *session, const char *name)
 	    name == NULL ? "" : name, name == NULL ? "" : ": ",
 	    "encountered an illegal file format or internal value");
 
-#if !defined(HAVE_DIAGNOSTIC)
-	return (__wt_panic(session));
+#if defined(HAVE_DIAGNOSTIC)
+	__wt_abort(session);                    /* Drop core if testing. */
 #endif
 
-	__wt_abort(session);			/* Drop core if testing. */
-	/* NOTREACHED */
+	return (WT_ERROR);
```

Comment:
@keithbostic, you should also take a look at this -- it changes `__wt_illegal_value` to not panic, which is necessary in MongoDB to get a decent error message out.  (One problem with panic is that as soon as MongoDB sees a `WT_PANIC` passed to its error handler, it shuts down).

I was actually a little surprised that this was a panic: are we sure we can never get here from verify or salvage, for example?